wagtail/snippets/views/snippets.py

```diff
@@ -363,45 +361,8 @@ class UsageView(generic.UsageView):
     view_name = "usage"
 
 
-class ActionColumn(Column):
-    cell_template_name = "wagtailsnippets/snippets/revisions/_actions.html"
-
-    def __init__(self, *args, object=None, view=None, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.object = object
-        self.view = view
-
-    def get_cell_context_data(self, instance, parent_context):
-        context = super().get_cell_context_data(instance, parent_context)
-        context["revision_enabled"] = isinstance(self.object, RevisionMixin)
-        context["draftstate_enabled"] = isinstance(self.object, DraftStateMixin)
-        context["preview_enabled"] = isinstance(self.object, PreviewableMixin)
-        context["can_publish"] = self.view.user_has_permission("publish")
-        context["object"] = self.object
-        context["view"] = self.view
-        return context
-
-
 class HistoryView(history.HistoryView):
     view_name = "history"
-    revisions_view_url_name = None
-    revisions_revert_url_name = None
-    revisions_compare_url_name = None
-    revisions_unschedule_url_name = None
-
-    @cached_property
-    def columns(self):
-        return [
-            ActionColumn(
-                "message",
-                object=self.object,
-                view=self,
-                classname="title",
-                label=_("Action"),
-            ),
-            UserColumn("user", blank_display_name="system"),
-            DateColumn("timestamp", label=_("Date")),
-        ]
```

Comment:
Moving all this to the generic view for a few reasons:

- It has very minimal to no impact to the readability/complexity to the generic HistoryView when used with models that don't support `RevisionMixin`/`DraftStateMixin`.
- It allows us to more easily share the code between generic models, pages, and snippets
- If in the future we want to support `RevisionMixin` with `ModelViewSet` (something that has come up a few times), the revision-related actions will be available immediately.